pymeasure/instruments/attocube/anc300.py

```diff
@@ -29,7 +29,45 @@
                                                      extract_float,
                                                      extract_int,
                                                      extract_value)
-from pymeasure.instruments.validators import strict_discrete_set, strict_range
+from pymeasure.instruments.validators import (joined_validators,
+                                              strict_discrete_set,
+                                              strict_range)
+
+
+def strict_length(value, values):
+    if len(value) != values:
+        raise ValueError(
+            'Value {} does not have an appropriate length of {}'.format(
+            value, values))
+    return value
+
+
+def truncated_int_array(value, values):
+    """ validator function to check if the supplied value is an iterable-object
+    with appropriate values.
+    """
+    ret = []
+    for i, v in enumerate(value):
+        if v < values[0]:
```

Comment:
Do you consider a value of -1.7 as OK? I.e. is the integer-ness only strict within the correct range?

Reply:
The device needs integer values from 0 to 255 (I guess one byte). I will change the logic to raise an error for out of range floats. good catch!

Reply:
yeah, your code already did that, it would clamp -1.7 to 0. 
The question was more, if a user supplied -1.7, do you consider this a "harmless" out of range (as you are ok with clamping to 0-255), or consider this a "grave" user error that should bail instead of clamping.